src/test/Cardano/CoinSelection/Algorithm/LargestFirstSpec.hs

```diff
@@ -245,3 +234,17 @@ propInputDecreasingOrder (CoinSelProp utxo txOuts) =
         $ selectCoins largestFirst
         $ CoinSelectionParameters utxo txOuts selectionLimit
     selectionLimit = CoinSelectionLimit $ const 100
+
+--------------------------------------------------------------------------------
+-- Utilities
+--------------------------------------------------------------------------------
+
+-- Returns true if (and only if) the given error value is one that can be
+-- thrown by the Largest-First algorithm.
+--
+isValidLargestFirstError :: CoinSelectionError -> Bool
```

Comment:
So we have still errors present that stopped to be valid in current impl of algo, right? Why not to prune them? We can always model those `exhausted` and `not-properly fragmented-inputs` to show that with new implementation they pass, no ?

Reply:
Or they are going to be used in `random` (ie., those errors) ? If not, I feel they are superfluous...

Reply:
@paweljakubas wrote:
>  So we have still errors present that stopped to be valid in current impl of algo, right? Why not to prune them? We can always model those exhausted and not-properly fragmented-inputs to show that with new implementation they pass, no ?

Good question!

Those errors are still used by the `Random-Improve` algorithm.

We **_could_** make it so that the algorithms return different sets of errors, but this would break backwards compatibility in the public API. (We'd need to increment the major version to do this.)

If we decide to do that, perhaps we should do it in a future PR. Thoughts?

Reply:
ok, if the errors are used in `random` then it seems ok to me. It even has additional  documentation aspect that the largest first does not deal with them expresses explicitely;-)

Reply:
> ok, if the errors are used in `random` then it seems ok to me. It even has additional documentation aspect that the largest first does not deal with them expresses explicitly;-)

Agreed! Though I do also agree we could consider making this more explicit in the types in a future version.